pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,134 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	windowsMetricsEndpoints          = "windows-machine-config-operator-metrics"
+	windowsMetricsEndpointsNamespace = "openshift-windows-machine-config-operator"
+	metricsPort                      = 9182
+)
+
+// prometheusConfig holds the information required to configure prometheus, so that it can scrape metrics from the
+// given endpoint address
+type prometheusConfig struct {
+	// k8sclientset holds the information related to kubernetes clientset
+	k8sclientset *kubernetes.Clientset
+	// hostname defines name of the target node.
+	hostname string
+	// endpointAddress holds the information about the endpoint address in the endpoints object
+	endpointAddress *v1.EndpointAddress
+}
+
+// NewPrometheusConfig creates a new instance for prometheusConfig  to be used by the caller.
+func NewPrometheusConfig(clientset *kubernetes.Clientset, ipAddress, hostname string) *prometheusConfig {
+	endpointAddress := &v1.EndpointAddress{
+		IP:        ipAddress,
+		TargetRef: &v1.ObjectReference{Kind: "Node", Name: hostname},
+	}
+	return &prometheusConfig{
+		k8sclientset:    clientset,
+		hostname:        hostname,
+		endpointAddress: endpointAddress}
+
+}
+
+// Configure patches the endpoint object to reflect the current list Windows nodes. When addIPAddress is
+// true it adds the endpoint address for the current node, else it removes that endpoint address
+func (pc *prometheusConfig) Configure(addIPAddress bool) error {
```

Comment:
Why does the caller need to decided if it is a situation of adding or removing an IP address? Shouldn't you be able to decide that based on the `endpointAddress`?

Reply:
`endpointAddress` just has the ipAddress and node info. The addition or removal of ip address depends on the events.

Given that we have a case where we need to reconfigure during pod restart, we would not be able to decide based on `endpointAddress`